scripts/code-check.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+set -euxo pipefail
+
+uv run ruff format . --check
+uv run ruff check .
+uvx yamllint .github
+
+if command -v shellcheck &> /dev/null; then
+  shellcheck scripts/*.sh
+else
+  echo "⚠️  shellcheck is not installed. Can't check shell scripts"
```

Comment:
Do you want to exit with a non-zero status here, or just make it a no-op?

Reply:
I thought a no-op was better, rather than require shellcheck to be installed. Don't mind though.